core/src/test/scala/shapeless/serialization.scala

```diff
@@ -937,11 +943,11 @@ class SerializationTests {
   def testLazy {
     assertSerializable(Lazy(23))
 
-    assertSerializable(implicitly[Lazy[Generic[Wibble]]])
-    assertSerializable(implicitly[Lazy[Generic1[Box, TC1]]])
+    assertSerializableBeforeAfter(implicitly[Lazy[Generic[Wibble]]])(_.value)
+    assertSerializableBeforeAfter(implicitly[Lazy[Generic1[Box, TC1]]])(_.value)
 
```

Comment:
Serialization doesn't force lazy vals. It just serializes the non-null fields. So if the lazy vals aren't materialized yet, their value won't be tested for serializability. I don't remember the exact details of how lazy vals are implemented, but I guess a closure get serialized in this case. A closure able to calculate the value if asked so.